### DISCUSSION TITLE
Problem with linker after update

### DISCUSSION THREAD

**Discussion by mikelekyr** [nanoframework/Home#1199] [General]
Hello everyone, I have a problem after updating ESP32, I am getting a linker error. First I updated my device using powershell:

![image](https://user-images.githubusercontent.com/121315809/209440784-230c5662-6285-4544-a9dc-b0ea2fefed1d.png)

I am using in my project these packages:

![image](https://user-images.githubusercontent.com/121315809/209440824-4927a984-8866-438c-8e62-b71a4d556aaf.png)

Some libraries are from about two days ago, but some of them are from october. Those older ones requre older library 'mscorlib' (1.12.0.4).
After these updates I am getting a Linker error. Please help me with this. Thank you. 

![image](https://user-images.githubusercontent.com/121315809/209440996-198632d2-4f32-430a-85ab-97723a26ac0a.png)

**Comment by alberk8** [comment]
Refer to this for the [cause.](https://discord.com/channels/478725473862549535/481780524864503828/1055508104244113458)

Possible [resolution.](https://discord.com/channels/478725473862549535/481780524864503828/1055508104244113458)  or [this.](https://discord.com/channels/478725473862549535/1056161355771686923)

**Reply by mikelekyr** [reply]
Do I have to register at the site you have sent the link to ?

**Reply by alberk8** [reply]
It is a Discord Server.   Join us [here](https://discord.gg/net-nanoframework-478725473862549535)

**Reply by mikelekyr** [reply]
Thank you, I was able to read the answers. Would you recommend me to wait for update of the other libraries, or downgrade the firmware? How can I do that, where can I find specific versions of firmware? Can I use the nanoff command to downgrade the firmware?

**Reply by josesimoes** [reply]
Yes you can specify a version when updating the firmware. Check the CLI options on the readme in the repo or wiht the -help option.

**Reply by mikelekyr** [reply]
For this task, I should be using nanoff app? I have not found anything in the help menu. Sorry, but I need help. Thank you.

**Reply by josesimoes** [reply]
I don't know which help menu you're referring to... Try `nanoff -h` for options.

Usage
https://github.com/nanoframework/nanoFirmwareFlasher#usage

**Reply by mikelekyr** [reply]
I was able to flash a previous version of firmware. The easiest would be to update the Nuget package for  nanoframework.Hardware.ESP32 and I could run on the newest firmware. All other nugets have been updated. Do you think it would be possible? Thank you.

**Reply by mikelekyr** [reply]
The current version is from October 21 and it is not compatible with the firmware.
